test/test_torch.py

```diff
@@ -5260,6 +5260,24 @@ def test_multiplication_numpy_scalar(self):
                 self.assertTrue(r2.dtype == t_dtype)
                 self.assertTrue(r2.requires_grad)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_parse_numpy_int(self):
+        self.assertRaisesRegex(RuntimeError, "Overflow",
+                               lambda: torch.mean(torch.randn(1, 1), np.uint64(-1)))
+        # https://github.com/pytorch/pytorch/issues/29252
+        for nptype in [np.int16, np.int8, np.uint8, np.int32, np.int64]:
+            np_arr = np.array([5], dtype=nptype)
+            np_val = np_arr[0]
+            self.assertEqual(torch.Storage(np_val).size(), 5)
```

Comment:
Regarding testing the new behavior:

I think we want to test the following things:
1) A numpy int parses correctly as a Long (int64) argument
2) A numpy int parses correctly for a Scalar argument (that it gets parsed as int64)
3) Some misc coverage.

What is currently in this test:
a) construct torch.Storage(np_int)
b) Assign `tensor[0]` a np_int
c) Add a `tensor` and an `np_int`. 

Case 2 is covered by (c). Constructing the storage and assigning a value falls into case (3) (because those aren't native functions and they have custom logic in their python bindings). We should have something for case 1 (something like torch.mean, that takes a long argument).

It would be good to add comments demonstrating what we're testing here. It's not clear to me as a reader how Pytorch parses `t + np_val` works, so some context that this is testing the np_int -> Scalar (Long) route might be nice, so something like:
```
// Test parsing numpy integer as pytorch Scalar gives a pytorch Long
self.assertEqual((t + np_val).dtype, t.dtype)
self.assertEqual((np_val + t).dtype, t.dtype)
```

Finally, I'm not sure if this is necessary, but maybe we could have some extra tests to check that the pytorch behavior is the same as the numpy behavior